apple/testing/default_runner/ios_test_runner.template.sh

```diff
@@ -284,3 +284,22 @@ if [[ -n "${COVERAGE_PRODUCE_JSON:-}" ]]; then
     exit 1
   fi
 fi
+
+if [[ -n "${COVERAGE_PRODUCE_HTML:-}" ]]; then
+  llvm_cov_html_export_status=0
+
+  # we couldn't use $COVERAGE_MANIFEST", as it will result an empty html
```

Comment:
After adding this here like this:

```
diff --git a/apple/testing/default_runner/ios_test_runner.template.sh b/apple/testing/default_runner/ios_test_runner.template.sh
index 67b44514..818d32d7 100644
--- a/apple/testing/default_runner/ios_test_runner.template.sh
+++ b/apple/testing/default_runner/ios_test_runner.template.sh
@@ -294,6 +294,7 @@ if [[ -n "${COVERAGE_PRODUCE_HTML:-}" ]]; then
     -format html \
     -use-color \
     "${lcov_args[@]}" \
+    @"$COVERAGE_MANIFEST" \
     > "$TEST_UNDECLARED_OUTPUTS_DIR/coverage.html"
     2> "$error_file" \
     || llvm_cov_html_export_status=$?
```

your test still passes for me, are you seeing something different? i think we always need to limit this otherwise you might get tons of transitive dependency noise in your html reports if you statically link your dependencies

Reply:
you are right, but adding `@"$COVERAGE_MANIFEST"` will result an empty HTML.

Reply:
@keith any update?

Reply:
I think the problem is that you need to `cd "$ROOT"` before doing this, does that work?

Reply:
@keith nice catch. so CMIIW, the `@COVERAGE_MANIFEST` will force the coverage to only include the files that are described there right?

I couldn't find any reference for `COVERAGE_MANIFEST` instead of this file, how do Bazel create this file?

Reply:
yes that sounds right, i don't see anything in the docs for it either. but you can `cat` it in the test script for debugging. IIRC the reason I hit this `cd` workaround before was because the relative paths were somehow invalid on disk from the default PWD in the test runner, which didn't work for `llvm-cov`. does

Reply:
yes, but i need to change the path for the xctest executable here, to get the full path on bazel-out
```swift
def _get_coverage_execution_environment(_ctx, covered_binaries):
    - covered_binary_paths = [f.short_path for f in covered_binaries.to_list()]
    + covered_binary_paths = [f.path for f in covered_binaries.to_list()]
```

Reply:
also looks like the `instrument_files` that is being used by `llvm-cov`, only contain file inside the unit test, so the HTML only consist of the source code for the unit test

Reply:
are you setting `--instrumentation_filter` to something that should include more binaries?

Reply:
@keith nope, just default bazel coverage

Reply:
I see the same issue here generally. Interestingly our internal test runner works with this just fine:

```sh
xcrun llvm-cov \
  show \
  -format html \
  -instr-profile "$profdata" \
  -path-equivalence=.,"$PWD" \
  -output-dir="$TEST_UNDECLARED_OUTPUTS_DIR/html" \
  "$test_tmp_dir/$test_bundle_name.xctest/$test_bundle_name" \
  @"$COVERAGE_MANIFEST"
```

so I'm still not sure why there's a difference.

Lets merge without the coverage manifest limiting today and try to fix it up later. Only other thing, should we do the html directory instead of a single file like my example here? that way you don't have 1 huge html file but many small ones and an index

Reply:
@keith regarding HTML file, AFAIK the `llvm-cov` only generate single HTML file.

Reply:
My example above outputs a directory

Reply:
@keith will update the PR

Reply:
@keith worked with `output-dir`. will it be better if we add an option to generate a single html file too?

Reply:
I'm not really sure. IMO it seems like folks with any non-trivial module size would want the directory instead, what do you think? looks like the test needs an update if we keep this route

Reply:
Agree, I think we can add it later if necessary.

will update the test

Reply:
@keith please help re review

Reply:
looks good but tests seem to be failing here

Reply:
is there any way to `cat` the log, to see the error detail?

Reply:
https://buildkite.com/bazel/rules-apple-darwin/builds/5897#01823cc1-6110-4617-9429-ad4db90aa7d3

Reply:
@thii yup, but the error log is located on a specific file

<img width="1400" alt="Screen Shot 2022-07-27 at 16 17 12" src="https://user-images.githubusercontent.com/16457495/181210885-7174fb64-e701-4a92-bb81-c0e3085ee35b.png">
e

Reply:
running it on my end seems to work, but I will try to mimic ci's command

Reply:
works on my end, but looks like on the ci
it tries to refer to `/app` and not `app`.

do you guys have any idea about it

<img width="2086" alt="Screen Shot 2022-07-28 at 15 28 44" src="https://user-images.githubusercontent.com/16457495/181459307-cf2da696-7955-4404-87e2-040e10ed2559.png">